src/lib/support/DefaultStorageKeyAllocator.h

```diff
@@ -190,6 +190,14 @@ class DefaultStorageKeyAllocator
 
     // Event number counter.
     static StorageKeyName IMEventNumber() { return StorageKeyName::FromConst("g/im/ec"); }
+
+    // Subscription resumption
+    static StorageKeyName FabricSubscription(FabricIndex fabric, NodeId nodeId)
```

Comment:
This assumes there can only be one stored. Is this true?

Reply:
FabricSubscription is "subscriptions for a node", so there'll be one for each fabric_node as seen in the construction of the key `"f/%x/su/%08"`.

SubscriptionResumptionIndex is the index and there's only one stored.

Reply:
OK, so this means that *all* subscriptions will be re-attempted. I thought there would be a way to make a single one canonical per fabric. Otherwise you are possibly trying to connect to N nodes on reboot, and maybe several of these subs didn't even intend to remain.

Reply:
The intent is to resume subscription when possible. Subscribers can reject them and get the subscription deleted for good. This is the same behavior as when a Publisher wakes up - it'll send report for each subscription it has.